src/content/en/2020/resource-hints.md

```diff
@@ -5,19 +5,367 @@ chapter_number: 21
+
+{{ figure_markup(
+  caption="Percentage of pages passing the offscreen images Lighthouse audit",
+  content="68.65%",
+  classes="big-number",
+  sheets_gid="1357389632",
+  sql_file="lighthouse_offscreen_images.sql"
+) }}
+
+Running Lighthouse's "[Defer offscreen images](https://web.dev/offscreen-images/)" audit resulted in 68.65% of pages passing the test. For those pages there is an opportunity to lazy-load images after all critical resources have finished loading.
+
+Be mindful to run the audit on both desktop and mobile as images may move off screen when the viewport changes.
+
+## Machine Learning
+
+Combining `prefetch` with Machine Learning can help with performance improvement of the subsequent page(s). One solution is [Guess.js](https://github.com/guess-js/guess) which made the initial breakthrough in predictive-prefetching, with over a dozen websites already using it in production.
+
+[Predictive prefetching](https://web.dev/predictive-prefetching/) is a technique that uses methods from data analytics and machine learning to provide a data-driven approach. Guess.js is a library that has predictive prefetching support for popular frameworks (Angular, Nuxt.js, Gatsby, and Next.js) and you can take advantage of it today. It ranks the possible navigations from a page and prefetches only the JavaScript that is likely to be needed next.
+
+Depending on the training set, the prefetching of Guess.js comes with over 90% accuracy.
+
+Overall, predictive prefetching is still uncharted territory but combined with prefetching on mouse over and even a Service Worker strategy, it has great potential to provide instant experiences for all users of the website, while saving their data.
+
+## HTTP/2 Push
+
+HTTP/2 has a feature called server push that can potentially improve page performance when your product experiences long [RTTs](https://developer.mozilla.org/en-US/docs/Glossary/Round_Trip_Time_(RTT)) or server processing. In brief, rather than waiting for the client to send a request, the server preemptively pushes a resource that it predicts the client will request soon afterwards.
+
+{{ figure_markup(
+  caption="Percentage of HTTP/2 Push pages using preload/nopush",
+  content="75.36%",
+  classes="big-number",
+  sheets_gid="308166349",
+  sql_file="h2_preload_nopush.sql"
+) }}
+
+In the 2020 dataset we have seen 1% of mobile pages using HTTP/2 Push, and of those 75% of preload header links use the nopush option in the firstHtml request. This means that even though a website is using preload, the majority of requests prefer to disable the push option.
```

Comment:
I think it's important to explain the relationship between preload and push.

```suggestion
HTTP/2 Push is often initiated through the `preload` link header. In the 2020 dataset we have seen 1% of mobile pages using HTTP/2 Push, and of those 75% of preload header links use the `nopush` option in the page request. This means that even though a website is using the `preload` resource hint, the majority prefer to use just this and disable HTTP/2 pushing of that resource.
```

Reply:
Niiice, yeah that's much better! Thank you